test/models/user_test.rb

```diff
@@ -320,6 +320,60 @@ def setup_user(operation)
     assert_includes record.errors.keys, :admin
   end
 
+  test "audit of password change should be saved only once, second time audited changes should not contain password_changed" do
+    user = FactoryGirl.create(:user)
+    user = User.find_by_id(user.id) #to clear the value of user.password
+    as_admin do
+      user.password = "newpassword"
+      assert_valid user
+      assert user.password_changed_changed?
+      assert user.password_changed
+      assert_includes user.changed, "password_changed"
+      assert user.save
+      #testing after_save
+      refute user.password_changed_changed?
+      refute user.password_changed
+      refute_includes user.changed, "password_changed"
+    end
+  end
+
+  test "audit of password change should be saved" do
+    user = FactoryGirl.create(:user)
+    user = User.find_by_id(user.id) #to clear the value of user.password
+    as_admin do
+      user.password = "newpassword"
+      assert_valid user
+      assert user.password_changed_changed?
+      assert user.password_changed
+      assert_includes user.changed, "password_changed"
+    end
+  end
+
+  test "audit of password change should not be saved - due to no password change" do
+    user = FactoryGirl.create(:user)
+    user = User.find_by_id(user.id) #to clear the value of user.password
+    as_admin do
+      user.firstname = "Johnny"
+      assert_valid user
+      refute user.password_changed_changed?
+      refute user.password_changed
+      refute_includes user.changed, "password_changed"
+    end
+  end
+
+  test "audit of name change sholud contain only firstname and not password_changed" do
```

Comment:
Yes that is it. I think is not necessary because if user.changed contains attribute, than Audit log contains it too. But I can change that if you want.